drivers/frequency/hmc7044/hmc7044.c

```diff
@@ -550,7 +551,11 @@ static int32_t hmc7044_setup(struct hmc7044_dev *dev)
 			continue;
 
 		hmc7044_write(dev, HMC7044_REG_CH_OUT_CRTL_0(chan->num),
-			      HMC7044_HI_PERF_MODE | HMC7044_SYNC_EN |
+			      (chan->start_up_mode_dynamic_enable ?
+			       HMC7044_START_UP_MODE_DYN_EN : 0) |
+			      (chan->output_control0_rb4_enable ? BIT(4) : 0) |
+			      (chan->high_performance_mode_dis ?
```

Comment:
chan->start_up_mode_dynamic_enable, chan->output_control0_rb4_enable and chan->high_performance_mode_dis are not initialized. hmc7044_init() should set them to the values passed through init_param.

Reply:
You are right. My bad.